From: Samanta Navarro <ferivoz@riseup.net>
Date: Tue, 15 Feb 2022 11:56:57 +0000
Subject: Prevent integer overflow in copyString
Origin: https://github.com/libexpat/libexpat/commit/efcb347440ade24b9f1054671e6bd05e60b4cafd
Bug: https://github.com/libexpat/libexpat/pull/560
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2022-25314

The copyString function is only used for encoding string supplied by
the library user.
[Salvatore Bonaccorso: Backport to 2.2.6 for context changes]
---
 expat/lib/xmlparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/expat/lib/xmlparse.c
+++ b/expat/lib/xmlparse.c
@@ -7446,7 +7446,7 @@ static XML_Char *
 copyString(const XML_Char *s,
            const XML_Memory_Handling_Suite *memsuite)
 {
-    int charsRequired = 0;
+    size_t charsRequired = 0;
     XML_Char *result;
 
     /* First determine how long the string is */
